go/vt/sqlparser/ast.go

```diff
@@ -712,36 +712,42 @@ type Union struct {
 	Into        *Into
 }
 
-// Union.Type
+// SetOp.Type
```

Comment:
would it be dumb to rewrite this into "SetOpType" and distinct as a bool? or does that lose information

Reply:
Not dumb, and it doesn't lose any information.
I just duplicated the existing logic; I'm pretty sure it was written that way in the first place so the yacc code is simpler.
Could just do
```sql
$$ = UnionStr
$$ = UnionAllStr
$$ = UnionDistinctStr
```
and convert it later instead of returning a new struct